src/controller/OperationalCredentialsDelegate.h

```diff
@@ -76,6 +76,12 @@ class DLL_EXPORT OperationalCredentialsDelegate
      *   fabric ID.
      */
     virtual void SetFabricIdForNextNOCRequest(FabricId fabricId) {}
+
+    virtual CHIP_ERROR GenerateNOCSR(MutableByteSpan & csrNonce)
+    {
+        ReturnErrorOnFailure(Crypto::DRBG_get_bytes(csrNonce.data(), csrNonce.size()));
```

Comment:
Check must be done that it's 32 bytes, and we must set the length of the `csrNonce` to 32 on success.

Reply:
See `kOpCSRNonceLength`, which should probably be moved to OperationalCredentialsDelegate.h